### DISCUSSION TITLE
Possible to compile just the engine?

### DISCUSSION THREAD

**Discussion by jonah-saltzman** [boa-dev/boa#2163] [Q&A] [answered]
Is it possible to strip boa down to just the engine, i.e. without WASM, CLI, profiler, tester, etc? I want to try compiling it for use in a C program.

**Comment by jedel1043** [comment, marked as answer]
`cargo build -p boa_engine` should generate a usable dynamic library of the pure engine.

However, we currently don't offer a compatibility layer with C, so you'll need to generate your own bindings by hand or by using [`cbindgen`](https://github.com/eqrion/cbindgen).

**Reply by Razican** [reply]
For anyone looking for a C API, this is being tracked here: #332
We currently lack resources to implement this, especially because the Rust API has not yet been defined in a 1.0 / stable way, but feel free to contribute!

**Comment by piaoger** [comment]
@Razican 
 
For my case, I am trying to embed boa into my rust app after reading the blog "Adding a JavaScript interpreter to your Rust project". After some quick tests, I realize it's really useful and even better if it's size can be smaller. 

Example:
An very simple app using boa is much larger than quickjs. 
```rust
    let mut context = boa_engine::Context::default();
    let res = context.eval("1+1").unwrap();
```

```rust
    let context = quick_js::Context::new().unwrap();
    let res = context.eval("1+1").unwrap();
```
boa:     10.9 MB
quickjs: 4.6 MB

**Reply by jedel1043** [reply]
Isn't that mainly because C libraries are generally smaller than Rust libraries without prior configuration? If you want to reduce the executable size, I'd recommend following https://github.com/johnthagen/min-sized-rust

**Reply by Razican** [reply]
It could be due to many things. What @jedel1043 mentions is just some of it. Probably panic unwinding (compared to the C aborting) will increase the size considerably.

It could also be related to the capabilities of the engine, how much text we use for errors, for example...

**Reply by piaoger** [reply]
@Razican  @jedel1043 
Thanks for your response. I just had a quick use. Binary size and even performance are not the most important factors at this stage.

For me, “Correctness” is more important. I have watched boa for long time and wonder if it can be used in some "not very critical" occasions, such as rule or scripting engine.I will have more tests on boa in next days. 

One question here, is the playground only based on stable version?
